py-polars/polars/dataframe/frame.py

```diff
@@ -7249,131 +7249,171 @@ def pivot(
+        the "Examples" section below.
 
         Parameters
         ----------
         values
-            Column values to aggregate. Can be multiple columns if the *columns*
-            arguments contains multiple columns as well. If None, all remaining columns
-            will be used.
+            The column(s) whose values will become the values of the output `DataFrame`.
+            If None, all columns not specified in the index and columns arguments will
+            be used.
         index
-            One or multiple keys to group by.
+            The column(s) whose values will become the index of the output `DataFrame`.
         columns
-            Name of the column(s) whose values will be used as the header of the output
-            DataFrame.
+            The column(s) whose values will become the columns of the output
+            `DataFrame`. If these columns are not :class:`String` columns, their values
+            will be coerced to strings, since polars column names must be strings.
         aggregate_function
-            Choose from:
-
-            - None: no aggregation takes place, will raise error if multiple values are in group.
-            - A predefined aggregate function string, one of
-              {'min', 'max', 'first', 'last', 'sum', 'mean', 'median', 'len'}
-            - An expression to do the aggregation.
+            A function to aggregate multiple `values` with the same `index` and
+            `columns` prior to pivoting, equivalent to using :func:`group_by` as a
+            preprocessing step. Choose from:
+
+            - `None`: no aggregation takes place; will raise an error if multiple values
+              are in a group.
+            - One of the strings {`'first'`, `'last'`, `'sum'`, `'max'`,
+              `'min'`, `'mean'`, `'median'`, `'len'`}, to perform predefined
+              types of aggregation.
+            - An expression that performs a custom aggregation, where
+              :func:`polars.element()` represents the multiple `values` in each "group"
+              with the same `index` and `columns`. For example, `aggregate_function='mean'`
+              is short for `aggregate_function=pl.element().mean()`.
         maintain_order
-            Sort the grouped keys so that the output order is predictable.
+            Whether to ensure that the row names of the output `DataFrame` are in the
+            same order they first appeared in the `index` of the input `DataFrame`.
+            `maintain_order=False` is not currently implemented.
         sort_columns
-            Sort the transposed columns by name. Default is by order of discovery.
+            Whether to order the non-index columns of the output `DataFrame` in
+            alphabetical order (if `sort_columns=True`), or in the same order they first
+            appeared in the `columns` of the input `DataFrame` (if `sort_columns=False`).
```

Comment:
not totally sure if the second half is an implementation detail, so would prefer to leave it out for now